### ISSUE TITLE
NOB will snap to an old position when changing ownership.

### ISSUE THREAD

**Issue by Disastercake** [FirstGearGames/FishNet#380]
**General**
Unity version: Unity 2021.3.22f1
Fish-Networking version: FishNetworking.3.5.8, FishNetworking.3.6.7.Pro, FishNetworking.3.7.2.Experimental.Pro
Link on our Discord where you troubleshot the issue(on Discord click the three dots next to the message, then Copy Message Link): 
https://discord.com/channels/424284635074134018/1110215219176816770/1110215219176816770

**Description**
When a NetworkObject ownership changes, it will snap to the position when it first changed ownership previously (for a single frame).

It seems like the position when it was owned is cached and used for one frame.  I imagine it's happening at the Fish Networking framework level. 

Notes:
- Was tested with Client Authoritative mode on for the NetworkTransform.
- Was tested with both NetworkTransform and a third party solution called SmoothSync as well, with the same results for both.

**Replication**

1. Open the attached test project.
2. Build the game and launch two instances.  Connect the instances using the default Network HUD.
3. On client: Press 'enter' key to request ownership and then move to another side of the screen (WASD keys).
5. On client: Press 'enter' key to request removal of ownership and watch the server's automation movement move the NOB.
6. On client: Press 'enter' to request ownership.
7. On server: Notice on server screen that the NOB quickly snaps to the position when the server first granted ownership.
8. On client: Move to another point in the screen (WASD keys) then remove ownership with the 'enter' key.
9. On client: Notice that the client screen snaps the NOB for one frame to the location when it first gained ownership.

**Expected behavior**
No snapping to occur and current owner's position to be respected immediately.

**Screenshots**
Video of the bug in a test project:
https://github.com/FirstGearGames/FishNet/assets/16954058/fc05bc4f-24da-4a53-ac4d-8f9c1af0a674

Test project in FishNetworking.3.5.8:
[FishNetBug_OwnerChangeSnap.zip](https://github.com/FirstGearGames/FishNet/files/11532440/FishNetBug_OwnerChangeSnap.zip)

**Comment by FirstGearGames** [comment]
I'll resolve this on the NT but you'll want to reach out to smoothsync
author if you want theirs fixed.

**Comment by Disastercake** [comment]
Okay thank you.  Once it's resolved on Fish Networking's end, I'll send them a diff of your changes so they know where to address the fix.

**Comment by FirstGearGames** [comment]
SmoothSync code should be nothing like my NT. I'm sure they have their own
code base. This isn't an issue outside the NT component.

**Comment by CNirvana** [comment]
I have a similar problem with NetworkTransform, may be it is related.

I have two players, one is the client-host, one is the client.
When I spawn an object from the object pool on the client-host side, the client side sees that the object appears first in the destroyed position in between and smoothen the transition to the current position.

Unity Version: 2021.3.20f1
FishNet Version: 3.6.5

```cs
Spawner
public class Spawner : NetworkBehaviour
{
    public NetworkObject prefab;
    public float spawnInterval = 0.5f;

    private float m_spawnTimer;

    private void Update()
    {
        if (IsServer)
        {
            m_spawnTimer -= Time.deltaTime;
            if (Input.GetKey(KeyCode.Space) && m_spawnTimer <= 0)
            {
                m_spawnTimer = spawnInterval;

                var bullet = NetworkManager.GetPooledInstantiated(prefab, 0, true);
                bullet.transform.SetPositionAndRotation(transform.position, transform.rotation);
                Spawn(bullet);
            }
        }
    }
}
```

Bullet
```cs
public class Bullet : NetworkBehaviour
{
    public float speed = 5.0f;

    private void Update()
    {
        if (IsServer)
        {
            var movement = speed * transform.forward * Time.deltaTime;
            if (Physics.Raycast(transform.position, transform.forward, out var hit, movement.magnitude, -1, QueryTriggerInteraction.Ignore))
            {
                Despawn();
                //gameObject.SetActive(false);
                return;
            }

            transform.position += movement;
        }
    }
}
```
[FishNetSandbox.zip](https://github.com/FirstGearGames/FishNet/files/11595290/FishNetSandbox.zip)

Here is a project that reproduces the issue.

You can reproduce this problem:
Launch two instances. One is Client-Host, One is Client.
On Client-Host side, press space key to fire some bullets.
On Client side,  you can see,  in some case, bullet smoothen from destroyed position to the current position.

**Comment by FirstGearGames** [comment]
The NetworkTransform received some updates due to this same behavior but with object pooling. This issue likely has the same reasoning as the pooling one, so I'd expect this to be resolved next update.

**Comment by FirstGearGames** [comment]
@Disastercake Going to send you a DM on discord because I cannot replicate in latest FN (in testing).

**Comment by FirstGearGames** [comment]
Resolved in 3.10.5.

**Comment by FirstGearGames** [comment]
Test copies were DisasterCake; did not receive a response on outcome yet. Issue is believed to be resolved in 3.10.5+ though.
If issue persist please re-open.
